quic/s2n-quic-core/src/connection/limits.rs

```diff
@@ -236,6 +238,18 @@ impl Limits {
         Duration
     );
     setter!(with_max_keep_alive_period, max_keep_alive_period, Duration);
+    /// Sets whether active connection migration is supported for a server endpoint (default: true)
+    ///
+    /// If set to false, the `disable_active_migration` transport parameter will be sent to the
+    /// peer, and any attempt by the peer to perform an active connection migration will be ignored.
+    pub fn with_connection_migration(mut self, enabled: bool) -> Result<Self, ValidationError> {
```

Comment:
Does it make sense to expose a non-exhaustive enum here instead if its functionally equivalent?

Imagining a wild future where this gets expanded to other values (maybe not):
- allow active mig (covered in this api)
- deny active mig (covered in this api)
- deny passive mig
- allow up to 5 active mig
- allow active mig from these IPs
- ..etc

Reply:
yeah, I went back and forth on that a bit. I think I ended up on thinking that the application shouldn't be too concerned about "passive" connection migration, so I left the "active" out of it. But thinking about it, its more obviously related to the `disable_active_migration` transport parameter with active in it, so I can add it

Reply:
Regarding the wild future you are talking about @toidiu, I think at that point we would have to stabilize the path migration validator and add more of the configuration in there. This one is really tied to the transport parameter, which doesn't have any of that configurability